utils/common/src/main/java/brooklyn/util/text/Strings.java

```diff
@@ -136,26 +150,45 @@ public static String removeFromEnd(String string, String ...suffixes) {
         return string;
     }
 
-    /** as removeFromEnd, but repeats until all such suffixes are gone */
-    public static String removeAllFromEnd(String string, String ...suffixes) {
+    /**
+     * As removeFromEnd, but repeats until all such suffixes are gone
+     */
+    public static String removeAllFromEnd(String string, String... suffixes) {
+        if (isEmpty(string)) return string;
+        int index = string.length();
         boolean anotherLoopNeeded = true;
         while (anotherLoopNeeded) {
             if (isEmpty(string)) return string;
```

Comment:
Yeah, but now `string` isn't modified so one check should be enough. I think that to keep the initial intent (to short circuit the loop if nothing is left of the string) the check should be replaced with `index == 0`.